### DISCUSSION TITLE
Push notifications

### DISCUSSION THREAD

**Discussion by SakuraIsayeki** [SakuraIsayeki/WOWS-Karma#103] [Ideas]
## Problem:
The current notifications system is extremely limited. Bound to a SignalR hub for communication, and only used on web app, we're seeing most users find limited use of that functionality, due to its limited action range.  
A user can only see notifications when in-app, and isn't notified when outside of the app.

##### See: https://en.wikipedia.org/wiki/Push_technology

We can provide users with a push notifications system, which would notify the user outside of the app, in a native fashion. Whether the user is using a desktop, mobile, or wearable, push notifications are only bound to the presence of a mainstream web browser (with current caveats only being limited to Apple blocking web push), and therefore can receive them even out of app, as a native app like notification.

This solution capitalises on the PWA features provided by the soon to be released Next web app ([app.wows-karma.com](https://app.wows-karma.com)).

## Notes:
 - **The complexity of the task is still to be defined**, as it touches advanced elements previously thought of outside our tech stack. With the Next app rewrite being made with Angular and TypeScript, we're bringing this field closer to our capabilities whilst expanding our stack.
- **The use of the notifications system is relative**, as it relies on existing users having setup notifications at least once before receiving any delivery through platform interactions. In other words, **this might just have a niche impact**.

**Comment by StealthPigeon** [comment]
I wouldnt use push notifications on either desktop or mobile, just by using the wows karma platform daily, the little "bell" notification area is more than enough to check up on new comments. But then again thats just a personal opinion
